tests/migration/reboot_to_upgrade.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Reboot machine to perform upgrade
+#       Just trigger reboot action, afterwards tests will be
+#       incepted by later test modules, such as tests in
+#       load_boot_tests or wait_boot in setup_zdup.pm
+# Maintainer: Qingming Su <qmsu@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    # Mark the hdd has been patched
+    set_var('PATCHED_SYSTEM', 1) if get_var('PATCH');
```

Comment:
why you need this variable ? Just call get_var('PATCH') everywhere where you need 'PATCHED_SYSTEM'

Reply:
The variables PATCH and PATCHED_SYSTEM are existing for different use: 
   - PATCH means the hdd will be patched, which is meanly used to load patching tests
   - PATCHED_SYSTEM means the hdd is already patched, which is used in lib/bootloader_setup.pm and tests/installation/bootloader_zkvm.pm to make sure s390x_zkvm machine reboot properly after patching: in patched systems just load the patched image, not copy hdd image again.

Reply:
agree